File: Swift Implementation/P7.swift
//
//  P7.swift
//  
//
//  Created by Xin Jet Chew on 08/01/2017.
//
//

import Foundation
import Darwin

func isAPrimeNum(n: Int) -> Bool {
    if n == 1 || (n != 2 && n % 2 == 0) {
        return false
    }
    else if n == 2 || n == 3 || n == 5 || n == 7 {
        return true
    }
    else {
        for i in 2 ... Int(sqrt(Double(n))) {
            if n % i == 0 {
                return false
            }
        }
    }
    return true
}


var numOfPrimes = 0
var i = 1

while(numOfPrimes < 10001) {
    if isAPrimeNum(n: i) {
        numOfPrimes += 1
        print("\(numOfPrimes): \(i) is a prime number")
    }
    i += 1
}

//<3

File: Java Implementation/src/P7.java
//10001st prime
public class P7 {

  private static final long VALUE = 600851475143L;

  static int[] primeNumbers = new int[60085147];
  static int counter = 1;

  public static void main(String[] args) {

    primeNumbers[0] = 2;

    for (int i = 3; counter <= 10001; i++) {
      if (isAPrimeNum(i)) {
        System.out.println("Prime number found: " + i);
        addToArray(i);
        if (counter == 10001) {
          System.out.println(i);
        }
      }
    }
  }

  public static boolean isAPrimeNum(int n) {
    for (int i = 0; i < counter; i++) {
      if ((n % primeNumbers[i]) == 0) {
        return false;
      }
    }
    return true;
  }

  public static void addToArray(int n) {
    for (int i = counter; i < VALUE; i++) {
      if (primeNumbers[i] == 0) {
        primeNumbers[i] = n;
        counter++;
        break;
      }
    }
  }


}
